src/components/views/read/PlayerView.tsx

```diff
@@ -97,9 +97,32 @@ const PlayerView: FC = () => {
     },
     referencesInfo: [],
   });
+  const directAssignSubkeys = [
+    'educationInfo',
+    'workInfo',
+    'skillsInfo',
+    'portfolioInfo',
+    'referencesInfo',
+  ];
 
-  const handleCvValuesChange = (newCvValues: CVInfoObj): void => {
-    setCvValues(newCvValues);
+  const handleCvValuesChange = (
+    subkey: string,
+    newSubkeyValues: Partial<CVInfoObj>,
+  ): void => {
+    if (directAssignSubkeys.includes(subkey)) {
```

Comment:
I was checking if the subkey passed is one value from the defined array above, so it will store it as array of objects not as object, so i will have this code right?
```
  const handleCvValuesChange = (
    subkey: keyof CVInfoObj,
    newSubkeyValues: Partial<CVInfoObj>,
  ): void => {
    setCvValues((prevCvValues) => ({
      ...prevCvValues,
      [subkey]: newSubkeyValues,
    }));
  };
```

Reply:
Mhmm, am I correct in saying that the Personal info section directly send the data to this function instead of updating the data once all info are collected ? In that case you should have all steps behave the same, sending the data once all info is completed and simply replace the value at the specific key in your `cvValues` state.

Reply:
That's right, and for other sections like education will be updated for each card, once Done button is clicked.

Reply:
For the function defined in the parent component, will it be like this, just to confirm that I'm on the right way.
```
 const handleCvValuesChange = (
    subkey: keyof CVInfoObj,
    newSubkeyValues: Partial<CVInfoObj>,
  ): void => {
    setCvValues((prevCvValues) => ({
      ...prevCvValues,
      [subkey]: {
        ...prevCvValues[subkey],
        ...newSubkeyValues,
      },
    }));
  };
```
If so, and I removed `[key: string]: any; // To access any object within this object.` from the types file, I will face an issue that is for example in the personalInfo section, I will get an error for not having the same type of the function we are using from the parent. `Type 'PersonalInfoObj' has no properties in common with type 'Partial<CVInfoObj>'`